walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,86 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+
+  private static final int START_OF_DAY = TimeRange.START_OF_DAY;
+  private static final int END_OF_DAY = TimeRange.END_OF_DAY;
+  private static final TimeRange WHOLE_DAY = TimeRange.WHOLE_DAY;
+  private static final Comparator<TimeRange> ORDER_BY_START = TimeRange.ORDER_BY_START;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    // Array of queries representing times to meet.
+    List<TimeRange> query = new ArrayList<>();
+
+    // Copy of events used for iterating and sorting.
+    List<Event> eventsCopy = new ArrayList<>(events);
+
+    int desiredDuration = (int) request.getDuration();
+
+    // Collection of string representing who is to attend the requested meeting.
+    Collection<String> requestAttendees = request.getAttendees();
+
+    // Check if duration of event is longer than a whole day and if so return empty.
+    if ((int) request.getDuration() > WHOLE_DAY.duration()) {
+      return query;
+    }
+
+    // Check if there are any events happening at all if so return whole day query.
+    if (events.size() == 0) {
+      // Adds a 24 hour time range to query.
+      query.add(TimeRange.WHOLE_DAY);
+      return query;
+    }
+
+    // Sorts the even in chronological order.
+    eventsCopy.sort((e1, e2) -> ORDER_BY_START.compare(e1.getWhen(), e2.getWhen()));
+
+    int currentTime = START_OF_DAY;
```

Comment:
This is never used outside the loop, so declaration and assignment can be joined below

Reply:
Ack